src/Process/MutantProcessContainer.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * This code is licensed under the BSD 3-Clause License.
+ *
+ * Copyright (c) 2017, Maks Rafalko
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Process;
+
+use function array_key_exists;
+use Infection\Process\Factory\LazyMutantProcessCreator;
+
+/**
+ * @internal
+ * @final
+ */
+class MutantProcessContainer
+{
+    /**
+     * @var list<MutantProcess>
+     */
+    private array $processes = [];
+
+    private int $currentProcessIndex = 0;
+
+    public function __construct(
+        MutantProcess $phpUnitMutantProcess,
+        /**
+         * @var list<LazyMutantProcessCreator>
+         */
+        private readonly array $lazyMutantProcessCreators,
+    ) {
+        $this->processes[] = $phpUnitMutantProcess;
+    }
+
+    public function getCurrentMutantProcessDetectionStatus(): string
```

Comment:
when reading this class it is clear to me that it is a holder of processes, hence I feel like those methods could be simplified:

- `getCurrentDetectionStatus()`
- `hasNext()`
- `createNext()` or even `next()`
- `getCurrent()`

This would make the API a bit lighter to read, and would also get closer to the PHP Iterator API which is kind of what this class is, an iterator for mutant processes